token/token.go

```diff
@@ -577,10 +577,15 @@ func looksLikeTimeValue(value string) bool {
 }
 
 // IsNeedQuoted whether need quote for passed string or not
-func IsNeedQuoted(value string) bool {
+func IsNeedQuoted(value string, ignoreIfMultiline bool) bool {
```

Comment:
It is strange that the change in encode option affects the token package, so if specified `LiteralStyle`, should fix in Encoder only.

For example, you can make changes to `(*Encoder).encodeString` .

Reply:
Thanks, addressed in [`47f32d7` (#164)](https://github.com/goccy/go-yaml/pull/164/commits/47f32d7bb037a5ba152fbafbbd3908f111e435d8)